### ISSUE TITLE
The UX of the HDMR's UGL

### ISSUE THREAD

**Issue by Golfavel** [pa1nki113r/Project_Brutality#945]
**Problem/Issues:**
The current underbarrel grenade launcher for the HDMR is fairly sound, however when it comes to the user experience regarding its state, it is fairly poor. Within a moment's glance it's nearly impossible to determine:
1. The weapon's current fire-mode (i.e., if pressing "fire" will shoot a grenade or a bullet)
2. If the launcher is loaded
3. (If you're not paying attention to whether the HUD is showing the "explosive" ammo counter) if you will even change your fire-mode, or begin to ADS.

**Solution(s):**
1. Adding a HUD element(s) to show all of the missing information mentioned
This would be the simplest solution, but also the least interesting; it would also lead to more work for other HUD developers.
2. Adding onto the weapon's sprite, essentially acting as the HUD element
This would mostly just be work for the artist(s); a seamless, glorified HUD element posing as a part of the weapon's sprite.
3. Changing how the underbarrel is interacted with
Currently, the way the underbarrel is used is a bit rudimentary, using secondary fire more as a quick weapon-switch. This could be streamlined to instead immediately fire the underbarrel, with then the development decision of either allowing the player to immediately fire the rifle itself again, or requiring the underbarrel to be reloaded immediately first, then switching back to rifle fire.
Allowing immediate rifle fire would allow for more action quickly and resolve issue 1, but would still require solution 1 or 2 to prevent issue 2 and 3 from remaining. Requiring the underbarrel to reload before switching back would fix both issue 1 and 2, but would make the experience slower and likely reduce its perceived usefulness, and would still require a solution for issue 3.

**Comment by JMartinez9820** [comment]
Yeah a lot of people have been complaining about the underbarrel grenade launcher function, but at this time it's not a top priority. We have been talking about making it use a similar system like BDv21 where it fires when pressing alt fire, or the grenade just simply reloads after firing. As for HUD, the cross hair changes determine if you're shooting bullets are grenades. There is a small green dot by the equipment counter showing that you fired a grenade.

**Comment by Golfavel** [comment]
Ah, I don't have a crosshair enabled so I never saw that difference.
